src/components/__tests__/Waveform.test.js

```diff
@@ -3,124 +3,184 @@ import { cleanup, fireEvent } from 'react-testing-library';
 import 'jest-dom/extend-expect';
 import Waveform from '../Waveform';
 import { renderWithRedux } from '../../services/testing-helpers';
-import Peaks from 'peaks';
-
-// Set up a redux store for the tests
-const peaksOptions = {
-  container: null,
-  mediaElement: null,
-  dataUri: null,
-  dataUriDefaultFormat: 'json',
-  keyboard: true,
-  _zoomLevelIndex: 0,
-  _zoomLevels: [512, 1024, 2048, 4096]
-};
 
+// Mock Peaks library
+const mockZoomIn = jest.fn();
+const mockZoomOut = jest.fn();
+const mockPlay = jest.fn();
+const mockPause = jest.fn();
+const mockPeaks = jest.genMockFromModule('peaks.js');
+mockPeaks.init = jest.fn(opts => {
+  return {
+    options: opts,
+    zoom: {
+      zoomIn: mockZoomIn,
+      zoomOut: mockZoomOut
+    },
+    player: {
+      play: mockPlay,
+      pause: mockPause
+    }
+  };
+});
+
```

Comment:
Mocked Peaks.js here, so that we can have mock functions and refer to them in the tests. And also I tried this with the other components, which turned out to be more code because some tests required more than a simple mock like `jest.fn()`. For an example `segments.getSegment(id)` needs to be implemented in a generic way because different tests in the same component calls it different times with different `ID`s. This resembled what we have currently.